nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/test/java/org/nuxeo/ecm/platform/picture/core/test/TestImagingConvertPlugin.java

```diff
@@ -146,40 +144,70 @@ public void testCrop() throws Exception {
         options.put(OPTION_RESIZE_WIDTH, cropWidth);
         options.put(OPTION_RESIZE_HEIGHT, cropHeight);
 
-        for (String filename : ImagingResourcesHelper.TEST_IMAGE_FILENAMES) {
-            String path = ImagingResourcesHelper.TEST_DATA_FOLDER + filename;
-            Blob blob = Blobs.createBlob(ImagingResourcesHelper.getFileFromPath(path));
-            blob.setFilename(filename);
-            BlobHolder bh = new SimpleBlobHolder(blob);
-
+        doOnTestImages((bh, path) -> {
             BlobHolder result = conversionService.convert(converter, bh, options);
             assertNotNull(result);
 
-            BufferedImage image = ImageIO.read(new FileInputStream(FileUtils.getResourceFileFromContext(path)));
+            BufferedImage image = readImage(path);
             assertNotNull("Original image is null", image);
             int width = image.getWidth();
             int height = image.getHeight();
             assertTrue("Original image size != (0,0)", width > 0 && height > 0);
 
-            image = ImageIO.read(result.getBlob().getStream());
+            image = readImage(result.getBlob());
             assertNotNull("Croped image is null", image);
             assertEquals("Croped image width", cropWidth, image.getWidth());
             assertEquals("Croped image height", cropHeight, image.getHeight());
-        }
+        });
     }
 
     @Test
     public void testConvertToPDF() throws Exception {
         String converter = "pictureConvertToPDF";
+        doOnTestImages((bh, path) -> {
+            BlobHolder result = conversionService.convert(converter, bh, null);
+            assertNotNull(result);
+        });
+    }
 
+    protected void doOnTestImages(BiConsumer<BlobHolder, String> consumer) throws IOException {
         for (String filename : ImagingResourcesHelper.TEST_IMAGE_FILENAMES) {
             String path = ImagingResourcesHelper.TEST_DATA_FOLDER + filename;
-            Blob blob = Blobs.createBlob(ImagingResourcesHelper.getFileFromPath(path));
-            blob.setFilename(filename);
-            BlobHolder bh = new SimpleBlobHolder(blob);
+            BlobHolder bh = getBlobHolder(path);
+            consumer.accept(bh, path);
+        }
+    }
 
-            BlobHolder result = conversionService.convert(converter, bh, null);
-            assertNotNull(result);
+    protected BlobHolder getBlobHolder(String path) throws IOException {
+        File file = ImagingResourcesHelper.getFileFromPath(path);
+        Blob blob = Blobs.createBlob(file);
+        MimetypeRegistry mimetypeRegistry = Framework.getService(MimetypeRegistry.class);
+        String mimeType = mimetypeRegistry.getMimetypeFromFilenameAndBlobWithDefault(file.getName(), blob, null);
+        blob.setMimeType(mimeType);
+        return new SimpleBlobHolder(blob);
```

Comment:
Can we use code from `AbstractConverterTest`?

Reply:
Unfortunately not :(